R/mlr_graphs_elements.R

```diff
@@ -0,0 +1,159 @@
+#' @include mlr_graphs.R
+
+#' @title Robustify a learner
+#' @name mlr_graphs_robustify
+#' @description
+#' Creates a [`Graph`] that can be used to robustify any subsequent learner.
+#' Performs the following steps:
+#' * Drops empty factor levels using [`PipeOpFixFactors`]
+#' * Imputes `numeric` features using [`PipeOpImputeHist`] and [`PipeOpMissInd`]
+#' * Imputes `factor` features using [`PipeOpImputeNewlvl`]
+#' * Encodes `factors` using `one-hot-encoding`. Factors with a cardinality > max_cardinality` are
+#'   collapsed using [`PipeOpCollapseFactors`].
+#' * If `scaling`, numeric features are scaled to mean 0 and standard deviation 1.
+#'
+#' The graph is built conservatively, i.e. the function always tries to assure everything works.
+#' If a learner is provided, some steps can be left out, i.e. if the learner can deal with
+#' factor variables, no encoding is performed.
+#'
+#' @param task [`Task`] \cr
+#'   A task to create a robustifying pipeline for.
+#'   Optional, if omitted, the full pipeline is created.
+#' @param learner [`Learner`] \cr
+#'   A learner to create a robustifying pipeline for. Optional, if omitted,
+#'   a more conservative pipeline is built.
+#' @param impute_missings [`logical`] | [`NULL`] \cr
+#'   Should missing values be imputed? Defaults to `NULL`, i.e imputes if the task has
+#'   missing values and the learner can not handle them.
+#' @param factors_to_numeric [`logical`] | [`NULL`] \cr
+#'   Should factors be encoded? Defaults to `NULL`, i.e encodes if the task has factors
+#'   and the learner can not handle factors.
+#' @param scaling [`logical`] \cr
+#'   Should the data be scaled? Defaults to `FALSE`.
+#' @param max_cardinality [`integer`] \cr
+#'   Maximum number of factor levels allowed. See above. Default: 1000.
+#' @return [`Graph`]
+#' @export
+#' @examples
+#' library(mlr3)
+#' lrn = lrn("regr.rpart")
+#' task = mlr_tasks$get("boston_housing")
+#' gr = robustify_pipeline(task, lrn) %>>% po("learner", lrn)
+#' resample(task, GraphLearner$new(gr), rsmp("holdout"))
+robustify_pipeline = function(task = NULL, learner = NULL, impute_missings = NULL,
+  factors_to_numeric = NULL, scaling = FALSE, max_cardinality = 1000) {
```

Comment:
I think we should remove scaling because (1) most methods that benefit from it have it built in already and (2) not scaling the data does not lead to crashed learners, only to bad models in some cases.